src/main/java/com/behrend/contestmanager/repository/UserRepository.java

```diff
@@ -0,0 +1,14 @@
+package com.behrend.contestmanager.repository;
+
+import com.behrend.contestmanager.models.User;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+
+public interface UserRepository extends CrudRepository<User, Long> {
+    //This helps with repository search when a user is logging in
+    @Query("SELECT u FROM User u WHERE LOWER(u.email) = LOWER(?1)")
+    User findByEmail(String email);
+
+    @Query("SELECT u FROM User u WHERE u.id = ?1")
```

Comment:
Both of these custom query strings should be unnecessary. Spring will auto-generate queries based on the function names, which these already fit the pattern for that.

Reply:
That's not what I am saying. Please read section 1.2.2 at this link:
https://docs.spring.io/spring-data/data-commons/docs/1.6.1.RELEASE/reference/html/repositories.html